src/frontend/src/scheduler/plan_fragmenter.rs

```diff
@@ -404,11 +405,38 @@ impl BatchPlanFragmenter {
                                 .is_none(),
                         "multiple table scan inside a stage"
                     );
-                    builder.table_scan_info = Some(TableScanInfo {
-                        vnode_bitmaps: table_desc
-                            .vnode_mapping
-                            .clone()
-                            .map(vnode_mapping_to_owner_mapping),
+
+                    builder.table_scan_info = Some({
+                        let table_desc = scan_node.logical().table_desc();
+
+                        let vnode_bitmaps =
```

Comment:
I'm not sure whether it's good to expose the vnode calculating stuffs to the frontend.🤔